ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java

```diff
@@ -138,40 +183,87 @@ private CacheConfiguration createCacheConfiguration(AssociationKeyMetadata assoc
 
 	private void appendIndex(QueryEntity queryEntity, AssociationKeyMetadata associationKeyMetadata, SchemaDefinitionContext context) {
 
-		if ( associationKeyMetadata.getColumnNames().length > 1 ) {
-			//composite id. not yet implemented
-			return;
+		for ( String idFieldName : associationKeyMetadata.getRowKeyColumnNames() ) {
+			queryEntity.addQueryField( generateIndexName( idFieldName ), String.class.getName(),null );
```

Comment:
Can you explain me what's going on here? How come this is always a String field?

Reply:
I generate fields for support association if it needs.

Reply:
It is part of method "appendIndex". It uses for create indexes for joins.

Reply:
I can see that but why does it pass String.class.getName as field type, couldn't the associaton field have some other type?